src/components/Tables/TableFilterAndExtraColumnsBar.tsx

```diff
@@ -73,18 +73,30 @@ function TableFilterAndExtraColumnsBar<Data, Filter extends FilterBase<Data, Sta
     const handleClear = (): void => {
         if (props.setFilters && props.filters) {
             props.setFilters(
-                props.filters?.map((filter) => ({
-                    ...filter,
-                    state: filter.initialState,
-                }))
+                props.filters?.map((filter) =>
+                    filter.persistOnClear
+                        ? {
+                              ...filter,
+                          }
+                        : {
+                              ...filter,
+                              state: filter.initialState,
+                          }
+                )
```

Comment:
The syntax `{ ...filter }` is (usually) equivalent to `filter`, so we can simplify this by writing:
```
                props.filters?.map((filter) =>
                    filter.persistOnClear
                        ? filter
                        : {
                              ...filter,
                              state: filter.initialState,
                          }
                )
```

And the same below :)